dbms/src/Functions/FunctionsExternalDictionaries.h

```diff
@@ -883,74 +843,19 @@ class FunctionDictGet final : public IFunction
         String attr_name = attr_name_col->getValue<String>();
 
         const auto id_col_untyped = block.getByPosition(arguments[2]).column.get();
-        const auto date_col_untyped = block.getByPosition(arguments[3]).column.get();
-        if (const auto id_col = checkAndGetColumn<ColumnUInt64>(id_col_untyped))
-            executeRange(block, result, dict, attr_name, id_col, date_col_untyped);
-        else if (const auto id_col = checkAndGetColumnConst<ColumnVector<UInt64>>(id_col_untyped))
-            executeRange(block, result, dict, attr_name, id_col, date_col_untyped);
-        else
-            throw Exception{"Third argument of function " + getName() + " must be UInt64", ErrorCodes::ILLEGAL_COLUMN};
-
-        return true;
-    }
+        const auto range_col_untyped = block.getByPosition(arguments[3]).column.get();
 
-    template <typename DictionaryType>
-    void executeRange(
-        Block & block, const size_t result, const DictionaryType * dictionary, const std::string & attr_name,
-        const ColumnUInt64 * id_col, const IColumn * date_col_untyped)
-    {
-        if (const auto date_col = checkAndGetColumn<ColumnUInt16>(date_col_untyped))
-        {
-            const auto size = id_col->size();
-            const auto & ids = id_col->getData();
-            const auto & dates = date_col->getData();
+        PaddedPODArray<UInt64> id_col_values_storage;
+        PaddedPODArray<Int64> range_col_values_storage;
+        const auto* id_col_values = getColumnDataAsPaddedPODArray(id_col_untyped, &id_col_values_storage);
+        const auto* range_col_values = getColumnDataAsPaddedPODArray(range_col_untyped, &range_col_values_storage);
```

Comment:
We apply conversion to conversion to Int64 in most cases.
This is Ok, because range dictionaries are not the most frequently used and performance-critical feature.
(it may lead to minor performance decrease)